lib/internal/fs/promises.js

```diff
@@ -65,13 +65,17 @@ const { promisify } = require('internal/util');
 const kHandle = Symbol('kHandle');
 const kFd = Symbol('kFd');
 const { kUsePromises } = binding;
+const {
+  JSTransferable, kDeserialize, kTransfer, kTransferList
+} = require('internal/worker/js_transferable');
 
 const getDirectoryEntriesPromise = promisify(getDirents);
 
-class FileHandle {
+class FileHandle extends JSTransferable {
```

Comment:
Does this have an impact on the memory consumption or size of `FileHandle` objects?

Reply:
According to the inspector, this adds 32 bytes in total, which seems to make sense to me (1 × vpointer + 3 × pointer-size fields in `BaseObject`).

For comparison, the `close` function on each `FileHandle` is 56 bytes in size.

Reply:
> For comparison, the close function on each FileHandle is 56 bytes in size.

That is actually pretty unfortunate to allocate a function per handle, but I guess maybe file handles are expected to be relatively expensive to hold?

Reply:
@benjamingr I would love to see our objects become smaller in general, but I think it’s fine here. There’s also the native `FileHandle` class, which also comes with 56 bytes per instance.

And as far as `FileHandle`s as a resource are concerned, we maybe create thousands of them in a single process, but not millions.